pkgs/test/lib/src/runner/browser/platform.dart

```diff
@@ -185,16 +185,23 @@ class BrowserPlatform extends PlatformPlugin
       var scriptBase = htmlEscape.convert(p.basename(test));
       var link = '<link rel="x-dart-test" href="$scriptBase">';
 
-      return shelf.Response.ok('''
-        <!DOCTYPE html>
-        <html>
-        <head>
-          <title>${htmlEscape.convert(test)} Test</title>
-          $link
-          <script src="packages/test/dart.js"></script>
-        </head>
-        </html>
-      ''', headers: {'Content-Type': 'text/html'});
+      if (_config.customHtmlTemplatePath != null) {
+        var contents = File(_config.customHtmlTemplatePath).readAsStringSync();
+        var processedContents = contents.replaceFirst('{testScript}', link);
```

Comment:
We may also want to allow for a test name placeholder `{{testName}}` (for title tags like below). This would not be required though like the `{{testScript}}` is.

This would also allow the default html file to be itself a template if you wish to do that but it isn't necessary.

Reply:
I'm not sure I understand the `This would also allow the default html file to be itself a template ` bit. You mean the HTML that is in the else branch, so it would be processed in the same way as the user provided, thus simplifying the logic?

Reply:
Yes exactly, it would simplify/unify the logic a bit. It would also be (negligibly) slower since it has to reach out to disk. You would likely use `Isolate.resolvePackageUri` to get the absolute file uri for the template file.

But again this is just a suggestion and not a requirement, I am fine with it as is :).